portfolio/src/main/webapp/script.js

```diff
@@ -29,8 +29,69 @@ function addRandomGreeting() {
   factContainer.innerText = "You found a hidden fact! \n \n" + fact;
 }
 
-async function getRandomQuoteUsingAsyncAwait() {
-  const response = await fetch('/data');
-  const quote = await response.text();
-  document.getElementById('quote-container').innerText = quote;
+// Fetches comments
+function getComments() {
+    console.log("Fetching comments")
+    const responsePromise = fetch('/data');
+    responsePromise.then(handleResponse);
 }
+
+// Handles response stream by converting it to text and passing the result to addQuoteToDom().
+function handleResponse(response) {
+  console.log('Handling the response.');
+  const textPromise = response.text();
+  textPromise.then(addCommentsToDom);
+}
+
+function addCommentsToDom(comments) {
+  console.log("Adding the following comments to DOM: " + comments);
+  const commentContainer = document.getElementById('comment-container');
+  commentContainer.innerText = comments;  
+}
+
+//take a text sectio and make it into a li element 
+function createListElement(text) {
+    const liElement = document.createElement('li');
+    liElement.innerText = text;
+    return liElement
+}
+
+// todo list 
+function loadTasks() {
+  fetch('/list-tasks').then(response => response.json()).then((tasks) => {
+    const taskListElement = document.getElementById('task-list');
+    tasks.forEach((task) => {
+      taskListElement.appendChild(createTaskElement(task));
+    })
+  });
+}
+
+// Creates an element that represents a task, including its delete button. 
+function createTaskElement(task) {
+  const taskElement = document.createElement('li');
+  taskElement.className = 'task';
+
+  const titleElement = document.createElement('span');
+  titleElement.innerText = task.title;
+
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteTask(task);
+
+    // Remove the task from the DOM.
+    taskElement.remove();
+  });
+
+  taskElement.appendChild(titleElement);
+  taskElement.appendChild(deleteButtonElement);
+  return taskElement;
+}
+
+// Tells the server to delete the task. 
+function deleteTask(task) {
+  const params = new URLSearchParams();
+  params.append('id', task.id);
+  fetch('/delete-task', {method: 'POST', body: params});
+}
```

Comment:
I don't see the servlet for these in this PR. I'd either add it to this PR or remove this from the PR,